msal4j-sdk/src/main/java/com/microsoft/aad/msal4j/ManagedIdentityApplication.java

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.microsoft.aad.msal4j;
+
+import com.nimbusds.oauth2.sdk.auth.ClientAuthentication;
+import lombok.Getter;
+import org.slf4j.LoggerFactory;
+
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * Class to be used to acquire tokens for managed identity.
+ * For details see {@link IManagedIdentityApplication}
+ * <p>
+ * Conditionally thread-safe
+ */
+public class ManagedIdentityApplication extends AbstractClientApplicationBase implements IManagedIdentityApplication {
```

Comment:
Should this be "final" ? in general, should new public API classes be final to prevent ppl from inheriting?

Reply:
I agree that this should be final. but the other public API classes are not. @Avery-Dunn maybe have this as part of refactoring?

Reply:
Is there are particular issue that could be cause by people inheriting this ManagedIdentityApplication class? As @neha-bhargava said none of the other classes are final.

From a strict Java best-practices standpoint unless there's a planned use case for inheritance then the class should be final, though it might be confusing for only this class to get singled out like that.

Reply:
We can add final to all the applications but do not want to do this as part of this PR.